app/controllers/navigation/Navigator.scala

```diff
@@ -255,6 +266,55 @@ object Navigator {
     case DocumentsProduced         => controllers.declaration.routes.AdditionalInformationController.displayPage
   }
 
+  val commonCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = Map.empty
+  val standardCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = Map.empty
+  val supplementaryCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = {
+    case ConsigneeDetails => consigneeDetailsSupplementaryPreviousPage
+  }
+  val simplifiedCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = Map.empty
+  val occasionalCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = Map.empty
+
+  val clearanceCacheDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode) => Call] = {
+    case ExporterDetails     => exporterDetailsClearancePreviousPage
+    case CarrierDetails      => carrierDetailsClearancePreviousPage
+    case ConsignorEoriNumber => consignorEoriNumberClearancePreviousPage
+    case RepresentativeAgent => representativeAgentClearancePreviousPage
+  }
+
+  private def exporterDetailsClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
+    if (cacheModel.isEntryIntoDeclarantsRecords)
+      controllers.declaration.routes.PersonPresentingGoodsDetailsController.displayPage(mode)
+    else
+      controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
+
+  private def carrierDetailsClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
+    if (cacheModel.parties.declarantIsExporter.exists(_.isExporter))
+      controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
+    else
+      controllers.declaration.routes.RepresentativeStatusController.displayPage(mode)
+
+  private def consigneeDetailsSupplementaryPreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
+    if (cacheModel.`type` == SUPPLEMENTARY && cacheModel.parties.declarantIsExporter.exists(_.isExporter))
+      controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
+    else
+      controllers.declaration.routes.RepresentativeStatusController.displayPage(mode)
+
+  private def consignorEoriNumberClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call = {
+    val declarantEori: Option[Eori] = cacheModel.parties.declarantDetails.flatMap(_.details.eori)
+    val exporterEori: Option[Eori] = cacheModel.parties.exporterDetails.flatMap(_.details.eori)
+
+    if (declarantEori.isDefined && exporterEori.isDefined && declarantEori == exporterEori)
+      controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
+    else
+      controllers.declaration.routes.ExporterDetailsController.displayPage(mode)
```

Comment:
This logic should be based on the answer Are you Exporter

Reply:
I am not the person to address this comment as I have only copied what was already there.
Seems we have a bug then.

Reply:
It has been fixed in https://github.com/hmrc/customs-declare-exports-frontend/pull/899/files

Reply:
I'll change it during rebasing